crates/iota-data-ingestion-core/src/worker_pool.rs

```diff
@@ -12,17 +12,29 @@ use iota_metrics::spawn_monitored_task;
 use iota_types::{
     full_checkpoint_content::CheckpointData, messages_checkpoint::CheckpointSequenceNumber,
 };
-use tokio::sync::{mpsc, oneshot};
+use tokio::sync::mpsc;
+use tokio_util::sync::CancellationToken;
 use tracing::info;
 
-use crate::{Worker, executor::MAX_CHECKPOINTS_IN_PROGRESS};
+use crate::{
+    Worker,
+    executor::{MAX_CHECKPOINTS_IN_PROGRESS, WorkerPoolMsg},
+};
 
 pub struct WorkerPool<W: Worker> {
     pub task_name: String,
     concurrency: usize,
     worker: Arc<W>,
 }
 
+#[derive(Debug, Clone, Copy)]
+enum WorkerMsg {
+    /// Send Worker progress status to WorkerPool main loop
+    Progress((usize, u64, Option<u64>)),
+    /// Signal Worker graceful shutdown to WorkerPool main loop
+    ShutDown(usize),
+}
```

Comment:
Similar to the suggestion in `executor`:

```rust
enum WorkerStatus {
    Running(_),
    Shutdown,
}
```
Please also consider making the variant description independent  more specific to the message's contents, as opposed to focusing on its calling scope.

Reply:
[fixed](https://github.com/iotaledger/iota/pull/4648/commits/f2d9f8da3328c5aae5815778d69daba4cdbd5bc9#diff-dfef1eeaea282ac0a41b60863611b2244808890824a073bf2264ca6fc775a624R30-R58)